DistroLauncher/WSLInfo.cpp

```diff
@@ -0,0 +1,133 @@
+//  along with this program.If not, see < https://www.gnu.org/licenses/>.
+//  .
+//  On Debian systems, the complete text of the GNU General
+//  Public License version 3 can be found in "/usr/share/common-licenses/GPL-3".
+//
+
+#include "stdafx.h"
+
+namespace Helpers {
+    namespace {
+        bool isX11UnixSocketMounted();
+        bool isWSLVDCPluginRegistered();
+    }
+
+    // Altough currently only used in this translation unit,
+    // this function has a lot of potential to not be exported.
+    bool WslLaunchSuccess(const TCHAR* command, DWORD dwMilisseconds) {
+        DWORD exitCode;
+        HANDLE child;
+        HRESULT hr = g_wslApi.WslLaunch(command, false, NULL, NULL, NULL, &child);
+        if (child == NULL || FAILED(hr)) {
+            return false;
+        }
+
+        WaitForSingleObject(child, dwMilisseconds);
+        auto success = GetExitCodeProcess(child, &exitCode);
+        CloseHandle(child);
+        if (success == FALSE || exitCode != 0) {
+            return false;
+        }
+
+        return true;
+    }
+
+    // Retrieves subsystem version from the WSL API.
+    DWORD WslGetDistroSubsystemVersion() {
+        ULONG distributionVersion;
+        ULONG defaultUID;
+        WSL_DISTRIBUTION_FLAGS wslDistributionFlags;
+        PSTR* defaultEnvironmentVariables;
+        ULONG defaultEnvironmentVariableCount;
+
+        HRESULT hr = g_wslApi.WslGetDistributionConfiguration(&distributionVersion,
+                                                              &defaultUID,
+                                                              &wslDistributionFlags,
+                                                              &defaultEnvironmentVariables,
+                                                              &defaultEnvironmentVariableCount);
+        if (FAILED(hr) || distributionVersion == 0) {
+            Helpers::PrintErrorMessage(hr);
+            return 0;
+        }
+
+        // discards the env variable string array otherwise they will leak.
+        for (int i = defaultEnvironmentVariableCount-1; i >= 0; --i) {
+            CoTaskMemFree((LPVOID)(defaultEnvironmentVariables[i]));
+        }
+
+        return distributionVersion;
+    }
+
+    bool isWslgEnabled() {
+        return isX11UnixSocketMounted() && isWSLVDCPluginRegistered();
+    }
+
+    bool WslGraphicsSupported() {
+        // Could WSL 3 or greater exist in the future?
+        if(Helpers::isWslgEnabled() &&
+            Helpers::WslGetDistroSubsystemVersion() > 1u) {
+            return true;
+        }
+
+        return false;
+    }
+     
+/* =========================== INTERNALS ================================== */
+    namespace {
+        // Returns true if the WSLDVCPlugin.dll is found.
+        // That is an indication that WSLg is installed.
+        // Search paths are empirical and subject to change.
+        bool isWSLVDCPluginRegistered() {
+            //TODO: if this hack survives, find more possible locations
```

Comment:
Yeah, this is why this is a little bit fragile IMHO, do we really need it or can we just rely on the socket symlink to not be created by WSL directly?

Reply:
My reasonsoning was to try to ensure Windows will be able to show the apps via RDP. But since Windows is the actor who launches wslg system distro and bind-mounts that folder, Windows having done all that and not being able to establish the RDP session would represent an invalid state way out of our control. So, having the socket listening should be enough for our purposes, I guess. I'll take that out.

Reply:
Done. `ls -l /tmp/.X11-unix` and `ss -lx | grep /tmp/.X11-unix/X0` both exiting 0 should be enough to believe WSLg is enabled.

Reply:
:+1: